services/notifications.go

```diff
@@ -1231,43 +1238,43 @@ func collectBlockProposalNotifications(notificationsByUserID map[uint64]map[type
 
 	for _, event := range events {
 		pubkey, err := GetGetPubkeyForIndex(event.Proposer)
-		if err == nil {
-			subscribers, ok := subMap[hex.EncodeToString(pubkey)]
-			if ok {
-				for _, sub := range subscribers {
-					if sub.UserID == nil || sub.ID == nil {
-						return fmt.Errorf("error expected userId or subId to be defined but got user: %v, sub: %v", sub.UserID, sub.ID)
-					}
-					if sub.LastEpoch != nil {
-						lastSentEpoch := *sub.LastEpoch
-						if lastSentEpoch >= epoch || epoch < sub.CreatedEpoch {
-							continue
-						}
-					}
-					logger.Infof("creating %v notification for validator %v in epoch %v", eventName, event.Proposer, epoch)
-					n := &validatorProposalNotification{
-						SubscriptionID: *sub.ID,
-						ValidatorIndex: event.Proposer,
-						Epoch:          epoch,
-						Status:         event.Status,
-						EventName:      eventName,
-						Reward:         event.ExecRewardETH,
-						EventFilter:    hex.EncodeToString(pubkey),
-					}
-					if _, exists := notificationsByUserID[*sub.UserID]; !exists {
-						notificationsByUserID[*sub.UserID] = map[types.EventName][]types.Notification{}
-					}
-					if _, exists := notificationsByUserID[*sub.UserID][n.GetEventName()]; !exists {
-						notificationsByUserID[*sub.UserID][n.GetEventName()] = []types.Notification{}
-					}
-					notificationsByUserID[*sub.UserID][n.GetEventName()] = append(notificationsByUserID[*sub.UserID][n.GetEventName()], n)
-					metrics.NotificationsCollected.WithLabelValues(string(n.GetEventName())).Inc()
+		if err != nil {
+			logger.Errorf("error retrieving pubkey for validator %v: %v", event.Proposer, err)
+			continue
+		}
+		subscribers, ok := subMap[hex.EncodeToString(pubkey)]
+		if !ok {
+			continue
+		}
+		for _, sub := range subscribers {
+			if sub.UserID == nil || sub.ID == nil {
+				return fmt.Errorf("error expected userId or subId to be defined but got user: %v, sub: %v", sub.UserID, sub.ID)
```

Comment:
Your error is "expect userId OR subId to be defined", but you check userID AND subID, as if one is nil, the error will be shown. So I guess the error is wrong, or the handling.